### ISSUE TITLE
RFE: Support creating instance from XtraBackup (API_RestoreDBInstanceFromS3)

### ISSUE THREAD

**Issue by jasonmcintosh** [terraform-providers/terraform-provider-aws#2410]
We'd like the ability to restore from an XtraBackup stored in an S3 bucket.  This is VERY fast creation of RDS databases vs. doing a load from a CSV file or similar.  I'm working on updating the code to support this myself, but wanted to present the issue here for reference later.

### Terraform Version
0.10.2

### Affected Resource(s)
aws_db_instance

### Terraform Configuration Files
Here's a conceptual of what this could look like... 
```hcl
resource "aws_db_instance" "myinstance" {
restore_s3_bucket "mybucket"
restore_s3_bucket_path "backups"
restore_s3_role "rolearn"
}
```
Option (I like this one better)
```hcl
resource "aws_db_instance" "myinstance" {
  restore_s3 {
    bucket  "mybucket"
    path "backups"
    role "restricted_s3_role_arn"
  }
}
```
### Important Factoids
Tied to...
http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Procedural.Importing.html
and API to implement is...
http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_RestoreDBInstanceFromS3.html

**Comment by jasonmcintosh** [comment]
Related btw to https://github.com/hashicorp/terraform/issues/16736

**Comment by bflad** [comment]
Support for `s3_import` argument on the `aws_db_instance` resource has been merged and will release with v1.16.0 of the AWS provider tomorrow 🎉

**Comment by bflad** [comment]
This has been released in [version 1.16.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1160-april-25-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
